packages/tpetra/CMakeLists.txt

```diff
@@ -13,6 +13,21 @@ TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 
 TRIBITS_SET_ST_FOR_DEV_MODE(ST_FOR_DEV_PT_FOR_RELEASE)
 
+# gcd 26 Oct 2018: Part of the great de-templating effort, 
+# allows users to switch at CMake time between using the old template
+# parameters, using Tpetra_DISABLE_DEPRECATED_CODE.  By default,
+# this will be set to "ON", to keep old behaviors.
+
+SET (Tpetra_DISABLE_DEPRECATED_CODE_DEFAULT OFF)
+MESSAGE (STATUS "Determine whether Tpetra will enable ordinal template arguments.")
+
+IF (Tpetra_DISABLE_DEPRECATED_CODE)
```

Comment:
Are you trying to define this option?  That's not how you do it.  Check out the definition of `Tpetra_ENABLE_CUDA` in `Trilinos/packages/tpetra/CMakeLists.txt` for an example.  I'll show you here how I would define `Tpetra_DISABLE_DEPRECATED_CODE`:
```
SET (Tpetra_DISABLE_DEPRECATED_CODE_DEFAULT OFF)

ASSERT_DEFINED (Tpetra_DISABLE_DEPRECATED_CODE_DEFAULT)
TRIBITS_ADD_OPTION_AND_DEFINE(
  Tpetra_DISABLE_DEPRECATED_CODE
  TPETRA_DISABLE_DEPRECATED_CODE
  "If ON, then disable deprecated code in Trilinos, at compile time."
  ${Tpetra_DISABLE_DEPRECATED_CODE_DEFAULT})

ASSERT_DEFINED (Tpetra_DISABLE_DEPRECATED_CODE)
IF (Tpetra_DISABLE_DEPRECATED_CODE)
  MESSAGE (STATUS "  - Tpetra_DISABLE_DEPRECATED_CODE is ON, so Tpetra will disable deprecated code at compile time.")
ELSE ()
  MESSAGE (STATUS "  - Tpetra_DISABLE_DEPRECATED_CODE is OFF, so Tpetra will not disable deprecated code at compile time.")
ENDIF ()
```

Reply:
Furthermore, this option is for all deprecated code, not just for deprecated template parameters.

Reply:
Remember that CMake has trinary logic:  `ON`, `OFF`, and `NOT DEFINED X`.  The latter doesn't quite mean the same thing as `OFF`.  This is why we take care to define the option before using it.

Reply:
Thanks, I added the support using TRIBITS_ADD_OPTION.